utils/navigate.py

```diff
@@ -0,0 +1,47 @@
+from utils import InstanceClassMethod
+from utils.log import logger
+
+
+class NavigateWrap(object):
+    def __init__(self, *args, **kwargs):
+        self.cls = args[0]
+
+    def __call__(self, f):
+        self.cls._nav_steps[f.__name__] = f
+        return f
+
+
+class Navigatable(object):
+    def __init__(self, *args, **kwargs):
+        self.navigate = self.instance_navigate
+
+    @InstanceClassMethod
+    def navigate(self, name):
```

Comment:
I am not that big fan of this approach, though I guess the class could get a param that will set that one func can get invoked in both contexts ...